projects/core/src/lib/tasks/task-list/task-list.component.ts

```diff
@@ -0,0 +1,117 @@
+        protected changeDetectorRef: ChangeDetectorRef,
+    ) {}
+
+    ngOnInit(): void {}
+
+    ngOnDestroy(): void {
+        this.taskSubscription?.unsubscribe();
+    }
+
+    ngAfterViewInit(): void {
+        this.taskSubscription = combineLatest({
+            sessionToken: this.userService.getSessionTokenStream(),
+            pageEvent: this.paginator.page.pipe(
+                startWith({
+                    pageIndex: this.paginator.pageIndex,
+                    pageSize: this.paginator.pageSize,
+                    length: this.paginator.length,
+                } as PageEvent),
+            ),
+        })
+            .pipe(
+                switchMap(({sessionToken, pageEvent}) => {
+                    this.isLoading = true;
+
+                    return forkJoin({
+                        pageEvent: of(pageEvent),
+                        tasks: this.backend.getTasksList(
+                            sessionToken,
+                            this.filter,
+                            pageEvent.pageIndex * pageEvent.pageSize,
+                            // we query one more to check if there is a next page
+                            pageEvent.pageSize + 1,
+                        ),
+                    });
+                }),
+                map(({pageEvent, tasks}) => {
+                    // Flip flag to show that loading has finished.
+                    this.isLoading = false;
+
+                    const newTotalResults = pageEvent.pageSize * pageEvent.pageIndex + tasks.length;
+                    this.paginator.length = Math.max(this.paginator.length, newTotalResults);
+
+                    // shrink to page since, since we queried one more
+                    return tasks.slice(0, pageEvent.pageSize);
+                }),
+            )
+            .subscribe({
+                next: (tasks) => {
+                    this.tasks = tasks;
+                    this.changeDetectorRef.markForCheck();
+                },
+            });
+    }
+
+    setFilter(taskStatus?: TaskStatusType): void {
+        this.filter = taskStatus;
+
+        // reset paginator and trigger reload
+        this.paginator.pageIndex = 0;
+        this.paginator.length = 0;
+
+        this.paginator.page.emit({
+            pageIndex: this.paginator.pageIndex,
+            pageSize: this.paginator.pageSize,
+            length: this.paginator.length,
+        } as PageEvent);
+    }
+
+    refreshPage(): void {
+        this.paginator.page.emit({
+            pageIndex: this.paginator.pageIndex,
+            pageSize: this.paginator.pageSize,
+            length: this.paginator.length,
+        } as PageEvent);
+    }
+
+    abortTask(taskId: UUID): void {
+        this.userService
+            .getSessionTokenForRequest()
+            .pipe(mergeMap((sessionToken) => this.backend.abortTask(sessionToken, taskId)))
+            .subscribe({
+                next: () => {
+                    this.refreshPage();
```

Comment:
the abort can fail, e.g. if the task was finished in the meantime. We should print show the error (and refresh anyway)